dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/SubscribeToEventTest.war/src/io/openliberty/org/apache/faces40/fat/subscribe/SubscribeTest.java

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.org.apache.faces40.fat.subscribe;
+
+import java.util.List;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.faces.component.UIComponent;
+import jakarta.faces.context.FacesContext;
+import jakarta.faces.event.AbortProcessingException;
+import jakarta.faces.event.ComponentSystemEvent;
+import jakarta.faces.event.ComponentSystemEventListener;
+import jakarta.faces.event.PreRenderViewEvent;
+import jakarta.faces.event.SystemEvent;
+import jakarta.faces.event.SystemEventListener;
+import jakarta.inject.Named;
+
+/**
+ * A test class that runs the applicable tests on the server to ensure SPEC behavior change for UIComponent#subscribeToEvent()
+ */
+@Named(value = "subscribeTest")
+@RequestScoped
+public class SubscribeTest {
+
+    public String getTestResults() {
+        //Test event
+        final Class<PreRenderViewEvent> event = PreRenderViewEvent.class;
+        final ComponentSystemEventListener myListener = new MyTestListener();
+
+        //Get context - shouldn't ever return null, but test just in case.
+        FacesContext ctx = FacesContext.getCurrentInstance();
+        if (ctx == null) {
+            return "false - FacesContext null";
+        }
+
+        //Get UIComponent - shouldn't ever return null, but test just in case.
+        UIComponent cc = UIComponent.getCurrentComponent(ctx);
+        if (cc == null) {
+            return "false - UIComponent null";
+        }
+
+        try {
+            //UIComponent#getListenersForEventClass() may never return null
+            testGetListenersForEventClass(cc, event);
+
+            //UIComponent#subscribeToEvent() must precheck if given listener is not already installed
+            for (int i = 1; i < 4; i++) {
+                try {
+                    testSubscribeToEvent(cc, event, myListener);
+                } catch (AssertionError ae) {
+                    throw new AssertionError("Failed testSubscribeToEvent after " + i + " attempt(s) with message: " + ae.getLocalizedMessage());
+                }
+            }
+
+            //UIComponent#unsubscribeToEvent should remove listeners
+            testUnsubscribeToEvent(cc, event, new MyTestListener());
```

Comment:
Should it be a new MyTestListener or the old one?  The comparison on here fails

https://github.com/apache/myfaces/blob/b2da263e6278c5a2df1b8b56f323122f99c008de/api/src/main/java/jakarta/faces/component/UIComponent.java#L836

Reply:
Too many tabs -- commented here https://github.com/OpenLiberty/open-liberty/pull/24129/files#r1101776666